tests/integration/tier0/backup-release/test_backup_release.py

```diff
@@ -1,37 +1,193 @@
 import os
-import platform
+import re
+
+from collections import namedtuple
+
+import pytest
 
 from envparse import env
 
 
-system = platform.platform()
+# TODO move to conftest
+class GetSystemVersion:
+    """
+    Helper class.
+    Assign a namedtuple with major and minor elements, both of an int type.
+    Assign a distribution (e.g. centos, oracle, rocky, alma0
+    Assign a system release.
+
+    Examples:
+    Oracle Linux Server release 6.10
+    Oracle Linux Server release 7.8
+    CentOS release 6.10 (Final)
+    CentOS Linux release 7.6.1810 (Core)
+    CentOS Linux release 8.1.1911 (Core)
+    """
+
+    with open("/etc/system-release", "r") as file:
+        system_release_content = file.read()
+        match_version = re.search(r".+?(\d+)\.(\d+)\D?", system_release_content)
+        if not match_version:
+            print("not match")
+        version = namedtuple("Version", ["major", "minor"])(int(match_version.group(1)), int(match_version.group(2)))
+        distribution = system_release_content.split()[0].lower()
+        system_release = "{}-{}.{}".format(distribution, version.major, version.minor)
 
 
-def test_backup_os_release_no_envar(shell, convert2rhel):
+class AssignRepositoryVariables:
     """
-    In this scenario there is no variable `CONVERT2RHEL_UNSUPPORTED_INCOMPLETE_ROLLBACK` set.
-    This means the conversion is inhibited in early stage.
-    This test case removes all the repos on the system which prevents the backup of some files.
-    Satellite is being used in all of test cases.
+    Helper class.
+    Assign correct repository setup to install subscription manager from to their respective distribution.
     """
 
+    # katello package associated variables
+    katello_pkg_url = "https://dogfood.sat.engineering.redhat.com/pub/katello-ca-consumer-latest.noarch.rpm"
+    katello_pkg_dst = "/usr/share/convert2rhel/subscription-manager/katello-ca-consumer-latest.noarch.rpm"
+
+    epel7_repository_setup = "[ubi]\nname=Universal Base Image 7\nbaseurl=https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi/server/7/7Server/x86_64/os/\ngpgcheck=0\nenabled=1"
```

Comment:
Cannot this also be made as a separate file that is then copied to the machine during the test run?

This is quite unreadable for me.